doc/releases/changelog-dev.md

```diff
@@ -111,7 +111,10 @@
 
 * Made `qml.OrbitalRotation` and consequently `qml.GateFabric` consistent with the interleaved Jordan-Wigner ordering.
   [(#3861)](https://github.com/PennyLaneAI/pennylane/pull/3861)
-
+  
+* `qml.devices.qubit.apply_operation` catches the `tf.errors.UnimplementedError` that occurs when `PauliZ` or `CNOT` gates
```

Comment:
i realize this isn't the most fitting changelog entry anymore... maybe let's sneak a touch-up into another PR?